js/searchbar/hostsSuggestionsPlugin.js

```diff
@@ -0,0 +1,45 @@
+var hosts = [];
+require('hostile').get(false, function (err, lines) {
+    if (!err) {
+        lines.forEach(function (line) {
+            line.forEach(function (host) {
+                host = host.split(' ');
+                host.forEach(function (host) {
+                    if (hosts.indexOf(host) === -1) {
+                        hosts.push(host);
+                    }
+                });
+            })
+        });
+    }
+});
+
+function showHostsSuggestions(text, input, event, container) {
+
+    empty(container);
+
+    var results = hosts.filter(function (host) {
+        return host.indexOf(text) > -1;
+    });
+
+    results.slice(0, 4).forEach(function (result) {
+
+        var item = createSearchbarItem({
+            title: result,
+            secondaryText: 'via hosts file',
+            url: 'http://'+result
+        });
+
+        container.appendChild(item);
+
+    })
+
+}
+
+registerSearchbarPlugin("hostsSuggestions", {
+    index: 1,
+    trigger: function (text) {
+        return !!text;
```

Comment:
There should probably be a minimum length to the text here, to avoid showing irrelevant entries. Once you visit a site a few times, the history search will start showing it immediately, so it probably isn't necessary to show these until a few characters have been typed.